portal/views/extend_flask_user.py

```diff
@@ -40,3 +44,46 @@ def reset_password_view_function(token):
 
     next_url = url_for('user.reset_password', token=token)
     return challenge_identity(user_id=user_id, next_url=next_url)
+
+
+class LockoutLoginForm(LoginForm):
+    """adds lockout functionality to the login process"""
+
+    def validate(self):
+        """prevent locked out users from logging in
+
+        Before verifying the user's credentials
+        check to see if the user is locked out.
+        If the user is locked out display an error
+        message below the password field.
+        """
+        # Find user by email address (email field)
+        user_manager = current_app.user_manager
+        user, user_email = user_manager.find_user_by_email(self.email.data)
+
+        # If the user is locked out display a message
+        # under the password field
+        if user.is_locked_out:
+            # Make sure validators are run so we
+            # can populate self.password.errors
+            super(LoginForm, self).validate()
+
+            auditable_event(
+                'local user attempted to login after being locked out',
+                user_id=user.id,
+                subject_id=user.id,
+                context='login'
+            )
+
+            error_message = _('We see you\'re having trouble - let us help. \
+                Your account will now be locked while we give it a refresh. \
+                Please try again in 30 minutes. \
```

Comment:
Can you replace the `30` with a variable (see below)? In we ever need to change the timeout value, it will save us the effort of re-translating it.

```
error_message = _('...Please try again in %d minutes...', 30)
```

Reply:
Oooo, good call